ntp.go

```diff
@@ -205,14 +243,30 @@ func getTime(host string, version int) (*msg, error) {
 		return nil, err
 	}
 
+	// It's possible to use random uint64 as client's `TransmitTime` field,
+	// it has better privacy (clock of the node is not disclosed in
+	// plain-text), better UDP packet spoofing resistance (blind attacker
+	// has to guess both port and the uint64 value), and OpenNTPD behaves
+	// like that. But math/rand is not secure enough for the purpose,
+	// crypto/rand takes 64 bits of entropy for every outgoing packet and
+	// CSPRNG from crypto/rand/rand_unix is not available: see
+	// https://github.com/golang/go/issues/13820
+	if m.OriginTime != xmt {
+		return nil, errors.New("response OriginTime != query TransmitTime") // spoofed packet?
+	}
+
 	return m, nil
 }
 
 // TimeV returns the current time from the remote server host using the
 // requested version of the NTP protocol. The version may be 2, 3, or 4;
 // although 4 is most typically used.
 func TimeV(host string, version int) (time.Time, error) {
-	m, err := getTime(host, version)
+	return TimeEx(host, QOption{Version: version})
+}
+
+func TimeEx(host string, opt QOption) (time.Time, error) {
```

Comment:
The `QueryWithOptions` function is the one people can use if they care about more advanced options.  I'd like to keep the `Time` functions as simple as possible.  So let's not use `TimeEx`.